x/blackbox/blackbox_test.py

```diff
@@ -0,0 +1,475 @@
+        "assertions": {
+            DRA.cost: lambda args: (
+                fib_cost(args) if args[0] < 17 else 70_000
+            ),
+            DRA.lastLog: lambda args: (
+                lightly_encode_output(fib(args[0]), logs=True)
+                if 0 < args[0] < 17
+                else None
+            ),
+            DRA.finalScratch: lambda args, actual: (
+                actual == {0: args[0], 1: fib(args[0])}
+                if 0 < args[0] < 17
+                else (True if args[0] >= 17 else actual == {})
+            ),
+            # we declare to "not care" about the top of the stack for n >= 17
+            DRA.stackTop: lambda args, actual: (
+                actual == fib(args[0]) if args[0] < 17 else True
+            ),
+            # similarly, we don't care about max stack height for n >= 17
+            DRA.maxStackHeight: lambda args, actual: (
+                actual == max(2, 2 * args[0]) if args[0] < 17 else True
+            ),
+            DRA.status: lambda args: "PASS" if 0 < args[0] < 8 else "REJECT",
+            DRA.passed: lambda args: 0 < args[0] < 8,
+            DRA.rejected: lambda args: 0 >= args[0] or args[0] >= 8,
+            DRA.noError: lambda args, actual: (
+                actual is True
+                if args[0] < 17
+                else "dynamic cost budget exceeded" in actual
+            ),
+        },
+    },
+}
+
+
+@pytest.mark.parametrize("filebase", APP_SCENARIOS.keys())
+def test_app_with_report(filebase: str):
+    mode, scenario = ExecutionMode.Application, APP_SCENARIOS[filebase]
+
+    # 0. Validate that the scenarios are well defined:
+    inputs, assertions = get_blackbox_scenario_components(scenario, mode)
+
+    algod = get_algod()
+
+    # 1. Read the TEAL from ./x/blackbox/teal/*.teal
+    path = Path.cwd() / "x" / "blackbox" / "teal"
+    case_name = filebase
+    tealpath = path / f"{filebase}.teal"
+    with open(tealpath, "r") as f:
+        teal = f.read()
+
+    print(
+        f"""Sandbox test and report {mode} for {case_name} from {tealpath}. TEAL is:
+-------
+{teal}
+-------"""
+    )
+
+    # 2. Build the Dryrun requests:
+    drbuilder = DryRunHelper.build_simple_app_request
+    dryrun_reqs = list(
+        map(lambda a: drbuilder(teal, lightly_encode_args(a)), inputs)
+    )
+
+    # 3. Run the requests to obtain sequence of Dryrun responses:
+    dryrun_resps = list(map(algod.dryrun, dryrun_reqs))
+
+    # 4. Generate statistical report of all the runs:
+    csvpath = path / f"{filebase}.csv"
+    with open(csvpath, "w") as f:
+        f.write(csv_from_dryruns(inputs, dryrun_resps))
+
+    # 5. Sequential assertions (if provided any)
+    for i, type_n_assertion in enumerate(assertions.items()):
+        assert_type, assertion = type_n_assertion
+
+        assert mode_has_assertion(
+            mode, assert_type
+        ), f"assert_type {assert_type} is not applicable for {mode}. Please REMOVE of MODIFY"
```

Comment:
@tzaffi Friendly pointer to what appears to be a minor typo.

```suggestion
        ), f"assert_type {assert_type} is not applicable for {mode}. Please REMOVE or MODIFY"
```